### PR TITLE
Overhauls the README, adds logo

### PR DESCRIPTION
**Pull request by cdrage**
This overhauls the README.md as well as adds a small logo to the top.
This includes some major changes to the grammar, wording as well as
format of the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,104 +1,151 @@
-# CentOS Community Container Pipeline
+![CentOS Community Container Pipeline](docs/logos/logo.png)
```
Comment (mohammedzee1000) [inline README.md]: +1 to this

**Comment by cdrage** [comment]
For a preview, please look here: https://github.com/cdrage/container-pipeline-service/tree/update-readme

File: README.md [hunk 2]
```diff
@@ -1,104 +1,151 @@
...
+- [atomic_scanners](atomic_scanners/misc-package-updates): Scans for outdated npm, pip and gem packages
 
-We're always looking for ideas and improvements for the service! If you're interested in contributing to this repository, follow these simple steps:
+After scanning and analysis, the resulting image is tagged and pushed to the internal registry as `image:rc`
 
-- open an issue on GitHub describing the feature/bug
-- fork the repository
-- work on your branch for the fix of the issue
-- raise a pull request
+6. **Push to the public registry (https://centos.registry.org)**
```
Comment (mohammedzee1000) [inline README.md]: Typo i think this should be registry.centos.org?

File: README.md [hunk 3]
```diff
@@ -1,104 +1,151 @@
...
+A developer wishing to host their container image will open up a pull request to the [CentOS Container Index](https://github.com/CentOS/container-index). 
+
+Once the pull request is merged, CCCP:
 
-1. **Project onboarding**
+  1. Links the Dockerfile
+  2. Builds the image
+  3. Scans / analyzes it
+  4. Pushes to [registry.centos.org](https://registry.centos.org)
+  5. Notifies the developer (email)
```
Comment (mohammedzee1000) [inline README.md]: +1 to step by step

File: README.md [hunk 4]
```diff
@@ -1,104 +1,151 @@
...
+
+2. **Jenkins and OpenShift tracking**
+
+Jenkins is utilized in order to track each application's Git repository as well as branch for any changes. This triggers a new build on **OpenShift** when a change is pushed.
+
+Changes to the application's repository, update to the base image or any RPMs that are part of the image will trigger a new build
+
+3. **Building the image**
+
+The container image is built within OpenShift and then pushed to OpenShift's internal registry. This results in an image tagged `image:test`.
```
Comment (mohammedzee1000) [inline README.md]: Originally, we used to push the image to Openshift internal registry. However, due to multiple issues, faced multiple times, for bringing up openshift internal registry, I believe we started to push to the public registry, with a temporary tag generated by hashing stuff like datetime

File: README.md [hunk 5]
```diff
@@ -1,104 +1,151 @@
...
+Scanning and analysis is done by using [Atomic](https://github.com/projectatomic/atomic).
 
-    An email is sent out the developer mentioning the status of the lint, build and scan processes and a link (s)he can use to read the detailed logs.
- 
-All the communication between the stages mentioned above happens via [beanstalkd](http://kr.github.io/beanstalkd/) tubes.   
+Multiple [Atomic Scanners](/atomic_scanners) are ran on the built image:
 
-## Contribute to Container Pipeline Service
+- [pipeline-scanner](atomic_scanners/pipeline-scanner): Scans for any outdated yum packages
+- [container-capabilities-scanner](atomic_scanners/container-capabilities-scanne): Scans for any security vulnerabitlies in the `RUN` command
```
Comment (mohammedzee1000) [inline README.md]: container-capabilities-scanner?

File: README.md [hunk 6]
```diff
@@ -1,104 +1,151 @@
...
+- [atomic_scanners](atomic_scanners/misc-package-updates): Scans for outdated npm, pip and gem packages
 
-We're always looking for ideas and improvements for the service! If you're interested in contributing to this repository, follow these simple steps:
+After scanning and analysis, the resulting image is tagged and pushed to the internal registry as `image:rc`
```
Comment (mohammedzee1000) [inline README.md]: Again, no internal registry is used. We push to public registry with a generated test tag.

File: README.md [hunk 7]
```diff
@@ -1,104 +1,151 @@
...
 
-## Setting up the Pipeline Service
+**Notes:**
 
-This will allow you to bring up a single or multi-node setup of Container Pipeline
-on various kinds of hosts (baremetal, a VPS, cloud or local VM, etc.) as long as they are accessible over SSH. This method uses Ansible for provisioning the hosts.
+All the communication between the stages mentioned above happens via [beanstalkd](http://kr.github.io/beanstalkd/) tubes.
 
-```bash
+## Architecture
```
Comment (mohammedzee1000) [inline README.md]: +1 for Architecture section

File: README.md [hunk 8]
```diff
@@ -1,104 +1,151 @@
...
+```sh
 # Provision the hosts. This assumes that you have added the usernames,
 # passwords or private keys used to access the hosts in the hosts file
-# above
-$ ansible-playbook -i hosts vagrant.yml
+$ ansible-playbook -i hosts main.yml
 ```
 
-## <a href="contact"></a>Get in touch
+## Contribute to the CentOS Community Container Pipeline Service
```
Comment (mohammedzee1000) [inline README.md]: +1 for this section

File: README.md [hunk 9]
```diff
@@ -1,104 +1,151 @@
...
+- work on your branch for the fix of the issue
+- raise a pull request
+
+Before a PR is merged, it must:
+
+- pass the CI done on [CentOS CI](https://ci.centos.org/)
+- be code reviewed by the maintainers
+- have maintainers' LGTM (Looks Good To Me)
+
+## Community
```
Comment (mohammedzee1000) [inline README.md]: +1 for this as well

**Comment by cdrage** [comment]
@mohammedzee1000 thanks for the review. updated!

File: README.md [hunk 10]
```diff
@@ -1,104 +1,151 @@
...
+How do I package that application into a container that's updated automatically every time I push changes? What about security and updates, how do I automate that each time I push any changes?
 
-The entire flow can be summarized as below
+That's where CCCP comes in. 
 
-![Container Pipeline Diagram](docs/diagrams/architecture.png)
+CCCP will:
+
+  - Scan the image for updates, fixes, capabilities and push it to a public registry (by default, http://registry.centos.org)
+  - Automatically rebuild when an RPM package is updated in the repository or base image (`FROM` in Dockerfile) is updated
```
Comment (bamachrn) [inline README.md]: Lets add 
* image update based on every push in source repo.
* also a line about weekly scan emails for notifying the user about scan results (here we do not update the image, we just notify the user about scan results)

File: README.md (hunk 1 above)
Comment (bamachrn) [reply README.md]: 👍

File: README.md [hunk 11]
```diff
@@ -1,104 +1,151 @@
...
+----
 
-## User Story
+CentOS Community Container Pipeline (CCCP) is an open-source platform to containerize applications based on CentOS.
 
-As an application developer, I want to develop applications on a stack (django, golang, nodejs, redis, rabbitmq, etc.) of my choice using CentOS as the base platform. I want to make sure that the application is packaged into a container and updated  automatically every time I push changes to my Git (GitHub, BitBucket, Gitlab, etc.) repository; the resulting container image is scanned for updates, fixes, capabilities, and delivered to a public registry from where the users can pull the image and run the application. I also want the container image to be automatically rebuilt when an RPM package is updated in the repository or base image (`FROM` in Dockerfile) is updated.
+**CCCP:** Builds the application (from a git repository) → Packages with the appropriate runtime → Tests and analyzes the image → Deploys to a public registry
```
Comment (bamachrn) [inline README.md]: do we say deploys to public registry? or uploads to public registry.

File: README.md [hunk 12]
```diff
@@ -1,104 +1,151 @@
...
+
+2. **Jenkins and OpenShift tracking**
+
+Jenkins is utilized in order to track each application's Git repository as well as branch for any changes. This triggers a new build on **OpenShift** when a change is pushed.
+
+Changes to the application's repository, update to the base image or any RPMs that are part of the image will trigger a new build
+
+3. **Building the image**
+
+The container image is built within OpenShift and then pushed to the registry. This results in an image tagged `image:test`.
```
Comment (bamachrn) [inline README.md]: image is not pushed to the registry.centos.org. its to the openshift internal one, I believe we are meaning the same here.
@cdrage how about taking you through the complete flow, for getting better understanding of pieces?

File: README.md [hunk 13]
```diff
@@ -1,104 +1,151 @@
...
 $ cd container-pipeline-service/provisions
 
 # Copy sample hosts file and edit as needed
 $ cp hosts.sample hosts
 ```
 
-You can have an all-in-one setup by using same value for all hosts in the `hosts` file or, use different hosts for each service.
+You can either have this span multiple-hosts or you can have an all-in-one setup by using the same host value in the `hosts` file.
+
+**An SSL certificate is required on the host running the registry:**
```
Comment (bamachrn) [inline README.md]: +1 for putting it all together.

File: README.md (hunk 12 above)
Comment (mohammedzee1000) [reply README.md]: so we are back to internal one cool :)

**Comment by cdrage** [comment]
@mohammedzee1000 @bamachrn Updated based on your changes.

**Comment by mohammedzee1000** [comment]
ok to test

**Pull request merged**